recipes/matio/all/conanfile.py

```diff
@@ -0,0 +1,102 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MatioConan(ConanFile):
+    name = "matio"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sourceforge.net/projects/matio/"
+    description = "Matio is a C library for reading and writing binary MATLAB MAT files."
+    topics = ("conan", "matlab", "mat-file", "file-format", "hdf5")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    exports_sources = "CMakeLists.txt", "patches/*"
+    options = {
+        "shared": [True, False],
+        "extended_sparse": [True, False],
+        "fPIC": [True, False],
+        "mat73": [True, False],
+        "with_hdf5": [None, "shared", "static"],
+        "with_zlib": [None, "shared", "static"]
```

Comment:
consumers can set the *sharedness* of the dependencies using `-o zlib:shared=True` etc.
```suggestion
        "with_hdf5": [True, False],
        "with_zlib": [True, False],
```